src/tests/functional/plugin/shared/include/shared_test_classes/subgraph/moe_builders.hpp

```diff
@@ -21,6 +21,7 @@ struct MoePatternParams {
     size_t topk;
     size_t number_of_experts;
     size_t intermediate_size;
+    bool with_gate_mul;  // whether add multiply on gate branch with constants, only applicale for MoE2GeMM.
```

Comment:
If it is applicable only for MoE2GeMM, let's throw an exception in case when with_gate_mul==true and moe type is MoE3GeMM